src/components/modal/report/modal.jsx

```diff
@@ -82,17 +97,25 @@ class ReportModal extends React.Component {
     render () {
         const {
             intl,
+            isOpen,
             onReport, // eslint-disable-line no-unused-vars
-            report,
+            onRequestClose,
+            reportStatus,
             type,
             ...modalProps
         } = this.props;
+        const isConfirmed = reportStatus === previewActions.Status.FETCHED;
+        const isError = reportStatus === previewActions.Status.ERROR;
+        const isWaiting = reportStatus === previewActions.Status.FETCHING;
```

Comment:
This is looking good! One further improvement would be to move this logic into `mapStateToProps` so the component would receive these as props, `isConfirmed`, `isError` and `isWaiting`.

Reply:
Done.